torchtext/data/example.py

```diff
@@ -9,6 +10,7 @@ class Example(object):
     """
     @classmethod
     def fromJSON(cls, data, fields):
+        warnings.warn('Example class will retire in 0.8.0 release and stay in torchtext.legacy. See 0.7.0 release note for the replacement.', UserWarning)
```

Comment:
I'd change the phrasing of this a bit.

"Example class will retire in 0.8.0 release and stay in torchtext.legacy. See 0.7.0 release note for the replacement"

to 

"Example class will be retired in the 0.8.0 release and moved to torchtext.legacy. Please see 0.7.0 release notes for further information."

Reply:
Did we settle on UserWarning? cc @gchanan

Reply:
With some offline discussions, `UserWarning` is proper, though pytorch started to use `FutureWarning`.